Python/traceback.c

```diff
@@ -517,9 +532,54 @@ tb_displayline(PyObject *f, PyObject *filename, int lineno, PyObject *name)
     Py_DECREF(line);
     if (err != 0)
         return err;
+    int truncation = _TRACEBACK_SOURCE_LINE_INDENT;
+    PyObject* source_line = NULL;
     /* ignore errors since we can't report them, can we? */
-    if (_Py_DisplaySourceLine(f, filename, lineno, 4))
+    if (!_Py_DisplaySourceLine(f, filename, lineno, _TRACEBACK_SOURCE_LINE_INDENT, &truncation, &source_line)) {
+        int code_offset = tb->tb_lasti;
+        if (PyCode_Addr2Line(_PyFrame_GetCode(frame), code_offset) != _PyCode_Addr2EndLine(_PyFrame_GetCode(frame), code_offset)) {
+            goto done;
+        }
+
+        Py_ssize_t start_offset = (Py_ssize_t) _PyCode_Addr2Offset(_PyFrame_GetCode(frame), code_offset);
+        Py_ssize_t end_offset = (Py_ssize_t) _PyCode_Addr2EndOffset(_PyFrame_GetCode(frame), code_offset);
+
+        if (start_offset < 0 || end_offset < 0) {
+            goto done;
+        }
+        if (end_offset == -1) {
+            // TODO: highlight from start_offset to the end of line
+            goto done;
```

Comment:
Now that thinking, I am not sure if we should do this. Since start might be somewhere in 250, and end might be somewhere in 260 with the total line being 300 chars. So it will annotate some irrelevant code as well. @pablogsal wdyt about dropping this?

Reply:
Hummmm, not sure I follow, could you drop an example?

Reply:
The `TODO: ` comment (AFAIK it was your idea) suggests that if `start_offset` is known but `end_offset` isn't (e.g start=250, end=260) we should start adding annotations from start_offset to `len(source_line)` instead of ignoring it for this case but that is wrong since if `len(source_line)` is `300` chars, then we will be annotating ~40 more chars then what we are supposed to. 


This is how it would look if we implement it:
```
long_call([245 chars], some_func_that_raises_an_error(arg), [50 chars])
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
Yeah I think if a line is longer than 255 characters then it's probably very long to begin with and printing from somewhere in the middle to the end could be misleading. Let's get rid of this TODO and not print carets if end information isn't available.

Reply:
AH, right, I remember now. Yeah, I agree with your analysis.